src/public.js

```diff
@@ -4,18 +4,21 @@
  */
 import isPublic from './helpers/isPublicPage.js'
 import isDocument from './helpers/isDocument.js'
+import isPdf from './helpers/isPdf.js'
+import NewFileMenu from './view/NewFileMenu.js'
 
 document.addEventListener('DOMContentLoaded', () => {
-
-	// Public share, but not a supported mimetype - do nothing
-	if (isPublic() && !isDocument()) {
+	if (!isPublic() || !OCA.Viewer) {
 		return
 	}
 
-	// Public share, and is a supported mimetype - open viewer
-	if (isPublic() && isDocument()) {
-		if (OCA.Viewer) {
-			OCA.Viewer.open({ path: '/' })
-		}
+	if (OCA.Files && OCA.Files.fileActions) {
+		OC.Plugins.register('OCA.Files.NewFileMenu', NewFileMenu)
+	}
+
+	if (isPdf()) {
```

Comment:
If I'm understanding you correctly, it *should* be fixed in my recent commits -- essentially, if the download is hidden *or* the files_pdfviewer is not enabled, open the PDF in richdocuments. Also open in richdocuments if it's an office document. otherwise, let the viewer handle it

Reply:
Yes, just went through testing the different constellations and looks all good 👍